script/tool/test/version_check_command_test.dart

```diff
@@ -514,6 +514,73 @@ void main() {
       );
     });
 
+    test(
+        'fails gracefully if the version headers are not found due to using the wrong style',
+        () async {
+      final Directory pluginDirectory =
+          createFakePlugin('plugin', packagesDir, version: '1.0.0');
+
+      const String changelog = '''
+## NEXT
+* Some changes for a later release.
+# 1.0.0
+* Some other changes.
+''';
+      createFakeCHANGELOG(pluginDirectory, changelog);
+      gitShowResponses = <String, String>{
+        'master:packages/plugin/pubspec.yaml': 'version: 1.0.0',
+      };
+
+      Error? commandError;
+      final List<String> output = await runCapturingPrint(runner, <String>[
+        'version-check',
+        '--base-sha=master',
+      ], errorHandler: (Error e) {
+        commandError = e;
+      });
+
+      expect(commandError, isA<ToolExit>());
+      expect(
+        output,
+        containsAllInOrder(<Matcher>[
+          contains('Unable to find a version in CHANGELOG.md'),
+          contains('The current version should be on a line starting with '
+              '"## ", either on the first non-empty line or after a "## NEXT" '
```

Comment:
How's this test passing without the space before the "##" above?

Reply:
(OK, it's [not passing](https://cirrus-ci.com/task/4581919208767488?logs=main#L302-L323) :P)

Reply:
I made a "cosmetic" line wrapping change just before pushing, and apparently didn't re-run tests.